### ISSUE TITLE
connection error in log

### ISSUE THREAD

**Issue by offerm** [ExchangeUnion/xud#463]
My node just started and learn about a peer node at 84.111.139.147:8885 from all there test nodes.
84.111.139.147:8885 is not reachable. I'm getting 3 errors on these instead of one (or zero actually).

Note that 84.11.1139.147 is my external IP address and I do run another XUD on 8885. 

I know why this is happaning. just think it is wrong. In any event, decide if this is a warning (like the log level) or and Error (like the message).

```
^Cadmins-MacBook-Pro-2:resolver admin$ ./start_xud.bash 
9/12/2018, 8:44:14 AM [GLOBAL] info: config loaded
9/12/2018, 8:44:14 AM [GLOBAL] info: Local nodePubKey is 024883dec76427e884e4c7f295578e1ed5dd5a9ebb91c6a2543c075ed9f35f7510
9/12/2018, 8:44:15 AM [LND] info: LndClient status: DISCONNECTED
9/12/2018, 8:44:15 AM [LND] info: trying to verify connection to lnd with uri: localhost:20002
9/12/2018, 8:44:15 AM [LND] info: LndClient status: DISCONNECTED
9/12/2018, 8:44:15 AM [LND] info: trying to verify connection to lnd with uri: localhost:20001
(node:43464) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
9/12/2018, 8:44:15 AM [LND] info: LndClient status: CONNECTION_VERIFIED
9/12/2018, 8:44:15 AM [LND] info: LndClient status: CONNECTION_VERIFIED
9/12/2018, 8:44:15 AM [P2P] info: retrieved external IP: 84.111.139.147
9/12/2018, 8:44:15 AM [P2P] info: p2p server listening on 0.0.0.0:8895
9/12/2018, 8:44:15 AM [P2P] info: Connecting to known / previously connected peers
9/12/2018, 8:44:15 AM [P2P] debug: Verifying reachability of advertised address: 84.111.139.147:8895
9/12/2018, 8:44:15 AM [RPC] info: gRPC server listening on localhost:7002
9/12/2018, 8:44:15 AM [P2P] debug: Connected pre-handshake to peer xud3.test.exchangeunion.com:8885
9/12/2018, 8:44:15 AM [P2P] debug: Connected pre-handshake to peer xud2.test.exchangeunion.com:8885
9/12/2018, 8:44:15 AM [P2P] debug: Connected pre-handshake to peer xud1.test.exchangeunion.com:8885
9/12/2018, 8:44:15 AM [P2P] info: Completed start-up connections to known peers.
9/12/2018, 8:44:16 AM [P2P] verbose: received 0 orders from 03fd337659e99e628d0487e4f87acf93e353db06f754dccc402f2de1b857a319d0
9/12/2018, 8:44:16 AM [P2P] verbose: received 0 orders from 028599d05b18c0c3f8028915a17d603416f7276c822b6b2d20e71a3502bd0f9e0a
9/12/2018, 8:44:16 AM [P2P] verbose: received 0 orders from 02b66438730d1fcdf4a4ae5d3d73e847a272f160fee2938e132b52cab0a0d9cfc6
9/12/2018, 8:44:25 AM [P2P] warn: Could not verify reachability of advertised address: 84.111.139.147:8895
9/12/2018, 8:44:26 AM [P2P] warn: error while opening connection to peer (0235584e2ac9a34a13ce6f6dc9fb8aa1e7918d7701f0b4b69dc002887e6b3fbba3): could not connect to peer at 84.111.139.147:8885: Connection timed out
9/12/2018, 8:44:26 AM [P2P] warn: error while opening connection to peer (0235584e2ac9a34a13ce6f6dc9fb8aa1e7918d7701f0b4b69dc002887e6b3fbba3): could not connect to peer at 84.111.139.147:8885: Connection timed out
9/12/2018, 8:44:26 AM [P2P] warn: error while opening connection to peer (0235584e2ac9a34a13ce6f6dc9fb8aa1e7918d7701f0b4b69dc002887e6b3fbba3): could not connect to peer at 84.111.139.147:8885: Connection timed out
```

**Comment by offerm** [comment]
@rsercano to reproduce this you need to have 2 XUDs running on your local device. Each one of them should connect to test1,2,3 servers (seed nodes). And you should connect them over localhost.

Once you do that and restart the local nodes you should see the problem.

If you fail to see it, please share the console log that you do see

**Comment by rsercano** [comment]
Could it be possible that I need an external IP address ? I see neither connection error nor timeout error if I use `localhost` or `internal ip`

```
9/13/2018, 4:37:16 PM [GLOBAL] info: config loaded
9/13/2018, 4:37:16 PM [GLOBAL] info: Local nodePubKey is 0356459563ec8b32419f113f011d639f9894c9e07c7d6dc99e61eab7173511aa7e
9/13/2018, 4:37:16 PM [DB] info: connected to database. host:localhost port:3306 database:xud
9/13/2018, 4:37:16 PM [LND] info: LndClient status: DISCONNECTED
9/13/2018, 4:37:16 PM [LND] info: trying to verify connection to lnd with uri: localhost:10009
9/13/2018, 4:37:16 PM [LND] info: LndClient status: DISCONNECTED
9/13/2018, 4:37:16 PM [LND] info: trying to verify connection to lnd with uri: localhost:10010
9/13/2018, 4:37:16 PM [RAIDEN] info: RaidenClient status: DISCONNECTED
9/13/2018, 4:37:17 PM [RAIDEN] error: could not get raiden address: 5.1 - raiden is disabled
(node:26497) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
9/13/2018, 4:37:17 PM [RAIDEN] error: Error: connect ECONNREFUSED 127.0.0.1:5001
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1162:14)
9/13/2018, 4:37:17 PM [LND] info: LndClient status: DISCONNECTED
9/13/2018, 4:37:17 PM [LND] error: could not verify connection to lnd at localhost:10009, error: {"code":14,"metadata":{"_internal_repr":{}},"details":"Connect Failed"}, retrying in 5000 ms
9/13/2018, 4:37:17 PM [LND] info: LndClient status: DISCONNECTED
9/13/2018, 4:37:17 PM [LND] error: could not verify connection to lnd at localhost:10010, error: {"code":14,"metadata":{"_internal_repr":{}},"details":"Connect Failed"}, retrying in 5000 ms
GELDİ false
9/13/2018, 4:37:17 PM [P2P] info: p2p server listening on 0.0.0.0:40544
9/13/2018, 4:37:17 PM [P2P] info: Connecting to known / previously connected peers
9/13/2018, 4:37:17 PM [P2P] debug: Verifying reachability of advertised address: xud1.test.exchangeunion.com:8885
9/13/2018, 4:37:17 PM [P2P] debug: Verifying reachability of advertised address: xud2.test.exchangeunion.com:8885
9/13/2018, 4:37:17 PM [P2P] debug: Verifying reachability of advertised address: xud3.test.exchangeunion.com:8885
9/13/2018, 4:37:17 PM [P2P] debug: Verifying reachability of advertised address: 10.3.11.60:8889
9/13/2018, 4:37:17 PM [RPC] info: gRPC server listening on localhost:3333
9/13/2018, 4:37:17 PM [P2P] warn: Could not verify reachability of advertised address: 10.3.11.60:8889
9/13/2018, 4:37:17 PM [P2P] debug: Connected pre-handshake to peer 35.229.81.83:8885
9/13/2018, 4:37:17 PM [P2P] debug: Connected pre-handshake to peer 35.196.118.79:8885
9/13/2018, 4:37:17 PM [P2P] warn: Could not verify reachability of advertised address: xud2.test.exchangeunion.com:8885
9/13/2018, 4:37:17 PM [P2P] warn: error while opening connection to peer (028599d05b18c0c3f8028915a17d603416f7276c822b6b2d20e71a3502bd0f9e0a): could not connect to peer at xud2.test.exchangeunion.com:8885: connect ECONNREFUSED 35.231.171.148:8885
9/13/2018, 4:37:17 PM [P2P] info: Completed start-up connections to known peers.
9/13/2018, 4:37:17 PM [P2P] warn: Could not verify reachability of advertised address: xud1.test.exchangeunion.com:8885
9/13/2018, 4:37:17 PM [P2P] warn: Could not verify reachability of advertised address: xud3.test.exchangeunion.com:8885
9/13/2018, 4:37:17 PM [P2P] verbose: received 0 orders from 03fd337659e99e628d0487e4f87acf93e353db06f754dccc402f2de1b857a319d0
9/13/2018, 4:37:17 PM [P2P] verbose: received 0 orders from 02b66438730d1fcdf4a4ae5d3d73e847a272f160fee2938e132b52cab0a0d9cfc6
9/13/2018, 4:37:22 PM [LND] info: trying to verify connection to lnd with uri: localhost:10009
9/13/2018, 4:37:22 PM [LND] info: trying to verify connection to lnd with uri: localhost:10010
9/13/2018, 4:37:22 PM [LND] info: LndClient status: DISCONNECTED
9/13/2018, 4:37:22 PM [LND] error: could not verify connection to lnd at localhost:10009, error: {"code":14,"metadata":{"_internal_repr":{}},"details":"Connect Failed"}, retrying in 5000 ms
9/13/2018, 4:37:22 PM [LND] info: LndClient status: DISCONNECTED
9/13/2018, 4:37:22 PM [LND] error: could not verify connection to lnd at localhost:10010, error: {"code":14,"metadata":{"_internal_repr":{}},"details":"Connect Failed"}, retrying in 5000 ms
9/13/2018, 4:37:27 PM [LND] info: trying to verify connection to lnd with uri: localhost:10009

```

**Comment by offerm** [comment]
No need to define any setting. Looks to me you are using advertise addresses.

You should be behind NAT.

I will try to reproduce myself later today or tomorrow morning.

**Comment by rsercano** [comment]
@offerm I still can't see the errors tried in a different environment, no luck. Probably as you said it's already solved.

**Comment by offerm** [comment]
Let's close it once the test nodes work properly and we give it another validation round.

**Comment by sangaman** [comment]
I see three logging enhancements mentioned here:

- Log how many nodes and how many new/valid nodes we receive from `NODES` packet.
- Reword the `warn` level messages so they don't use the word "error" in the actual message.
- Log a message on successful handshake that we got a nodepubkey from a given addresss/port so we know which nodepubkey belongs to which ip address from the logs.

I'm not exactly sure what the root problem is here though. You're advertising an address that can't be reached externally?

**Comment by offerm** [comment]
Agree with the list of enhancements.  

The problem I reported here was the parallel 3 attempts to connect to 
(0235584e2ac9a34a13ce6f6dc9fb8aa1e7918d7701f0b4b69dc002887e6b3fbba3):  84.111.139.147:8885

I'm not sure that the issue was resolved since the test servers are not really "working" now.

From here is just FYI:
Between the lines, I also complained about the current functionality of detecting my external IP  84.111.139.147 as my listen address and adding that to the XUD test servers upon connecting to them.

This resulted by my two localhost XUDs trying to connect each other over 84.111.139.147. They can't do that since they are behind firewall.


Anyway, this was discussed in long and there is a plan of action.
